volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,374 @@
+        saddr_list.append(src_portid)
+        if netlink_sock.groups != 0:
+            groups_bitmap = netlink_sock.groups.dereference()
+            groups_str = f"groups:0x{groups_bitmap:08x}"
+            saddr_list.append(groups_str)
+
+        daddr_list = []
+        dst_portid = f"portid:{netlink_sock.dst_portid}"
+        daddr_list.append(dst_portid)
+        dst_group = f"group:0x{netlink_sock.dst_group:08x}"
+        daddr_list.append(dst_group)
+        module = netlink_sock.module
+        if module and netlink_sock.module.name:
+            module_name_str = utility.array_to_string(netlink_sock.module.name)
+            module_name = f"lkm:{module_name_str}"
+            daddr_list.append(module_name)
+
+        saddr_tag = ",".join(saddr_list)
+        daddr_tag = ",".join(daddr_list)
+        state = netlink_sock.state
+
+        sock_stat = saddr_tag, daddr_tag, state
+        return netlink_sock, sock_stat
+
+    def _vsock_sock(self, sock, _extended):
+        vsock_sock = sock.cast("vsock_sock")
+        saddr = vsock_sock.local_addr.svm_cid
+        sport = vsock_sock.local_addr.svm_port
+        daddr = vsock_sock.remote_addr.svm_cid
+        dport = vsock_sock.remote_addr.svm_port
+        state = ""  # Protocol is always 0
+
+        saddr_tag = f"{saddr}:{sport}"
+        daddr_tag = f"{daddr}:{dport}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return vsock_sock, sock_stat
+
+    def _packet_sock(self, sock, extended):
+        packet_sock = sock.cast("packet_sock")
+        ifindex = packet_sock.ifindex
+        dev_name = self._netdevices.get(ifindex, "") if ifindex > 0 else "ANY"
+
+        if sock.has_member("sk_filter"):
+            sock_filter = sock.sk_filter
+            self.__update_extra_socket_bpf(sock_filter, extended)
+
+        if sock.has_member("sk_reuseport_cb"):
+            sock_reuseport_cb = sock.sk_reuseport_cb
+            self.__update_extra_socket_bpf(sock_reuseport_cb, extended)
+
+        saddr_tag = f"{dev_name}"
+        daddr_tag = ""
+        state = packet_sock.state
+        sock_stat = saddr_tag, daddr_tag, state
+        return packet_sock, sock_stat
+
+    def __update_extra_socket_bpf(self, sock_filter, extended):
+        if not sock_filter:
+            return
+
+        extended["bpf_filter_type"] = "cBPF"
+
+        if not sock_filter.has_member("prog"):
+            return
+
+        bpfprog = sock_filter.prog
+        if not bpfprog:
+            return
+
+        BPF_PROG_TYPE_UNSPEC = 0
+        if bpfprog.type > BPF_PROG_TYPE_UNSPEC:
+            extended["bpf_filter_type"] = "eBPF"
+            bpfprog_aux = bpfprog.aux
+            if bpfprog_aux:
+                extended["bpf_filter_id"] = str(bpfprog_aux.id)
+                bpfprog_name = str(utility.array_to_string(bpfprog.aux.name))
+                if bpfprog_name:
+                    extended["bpf_filter_name"] = bpfprog_name
+
+    def _xdp_sock(self, sock, _extended):
```

Comment:
What's the point in the unnamed parameter `_extended`?  If it's just to match the other calls that's ok, but docstrings would go a long way to clarifying that...